### PR TITLE
Update schema to validate breaking change in metrics

### PR DESCRIPTION
**Pull request by sudanl0**
## Change:
Update the Firecracker metrics schema to cover breaking change in
LatencyAggregateMetrics.
Use the 'additionalProperties' flag to catch new metrics which are not
covered by jsonschema validate and add required missing metrics to the schema.

## Reason:
LatencyAggregateMetrics added nested dictionaries in Firecracker's
json-formatted-metrics-object and validate_fc_metrics() failed to cover
breaking change in these metrics.
validate_fc_metrics() reported block_rootfs and vcpu metrics missing
from jsonschema validation after the additionalProperties flag was added.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this
  PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [x] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/host_tools/fcmetrics.py [hunk 1]
```diff
@@ -33,6 +33,12 @@ def create_metrics_schema_objects(metrics):
     }
 
     if isinstance(metrics, dict):
+        special_metrics = "utc_timestamp_ms"
```
Comment (kalyazin) [inline tests/host_tools/fcmetrics.py:36]: commit message `fix(test): Update metrics scheme to cover missing metrics`:

```diff
-added to the schema but, add it now to keep jsconschema.validate happy.
+added to the schema but, add it is now to keep jsonschema.validate happy.
```

File: tests/host_tools/fcmetrics.py [hunk 2]
```diff
@@ -276,29 +309,7 @@ def validate_fc_metrics(metrics):
...
-                metrics_name in latency_agg_metrics
-                and metrics_field in latency_agg_metrics[metrics_name]
-            ):
-                metrics_type = "object"
-            else:
-                metrics_type = "number"
-            metrics_schema["properties"][metrics_field] = {"type": metrics_type}
-        firecracker_metrics_schema["properties"][metrics_name] = metrics_schema
-        firecracker_metrics_schema["required"].append(metrics_name)
+    firecracker_metrics_schema = create_metrics_schema_objects(firecracker_metrics)
```
Comment (kalyazin) [inline tests/host_tools/fcmetrics.py:329]: nit: would it be sensible to add a "test for test" to make sure that it captures missing metrics? Maybe a simple smoke test with a very sparse input structure to check that the validation rejects it.

Comment (sudanl0) [reply tests/host_tools/fcmetrics.py:329]: initially I though about adding some negative test cases but, after adding the "additionalProperties:false" flag in the schema, the test started reporting all the missing metrics including the block/net and latency aggregate. So, I think there is no need for negative tests.

File: tests/host_tools/fcmetrics.py (hunk 1 above)
Comment (kalyazin) [reply tests/host_tools/fcmetrics.py:36]: update:
commit message `fix(test): Update metrics scheme to cover missing metrics`:

```diff
-added to the schema but, add it now to keep jsconschema.validate happy.
+added to the schema but, add it now to keep jsonschema.validate happy.
```

Comment (sudanl0) [reply tests/host_tools/fcmetrics.py:36]: done, thanks!

File: tests/host_tools/fcmetrics.py (hunk 2 above)
Comment (kalyazin) [reply tests/host_tools/fcmetrics.py:329]: fair enough, `additionalProperties:false` looks convincing.

**Review by kalyazin — APPROVED** [review]

**Review by pb8o — APPROVED** [review]

**Pull request merged**